### ISSUE TITLE
May I ask if other logging frameworks can be used together with POEM

### ISSUE THREAD

**Issue by 525bigword** [poem-web/poem#636]
May I ask if other logging frameworks can be used with POEM, such as log4rs, to save logs to a local file

**Comment by sunli829** [comment]
Yes, this is not a conflict, you can implement a middleware to output logs, like this:

https://github.com/poem-web/poem/blob/master/poem/src/middleware/tracing_mw.rs

**Comment by 525bigword** [comment]
> Yes, this is not a conflict, you can implement a middleware to output logs, like this:
> 
> https://github.com/poem-web/poem/blob/master/poem/src/middleware/tracing_mw.rs

`
let app = Route::new().at("/registry", post(registry))
                    .at("/info", get(registry_info))
                    .with(Tracing)
`
I don't know if my usage is correct. If I want to use log4rs, how do I need to use it,After adding. .with (Tracing), there was no active output of any logs

**Comment by sunli829** [comment]
As far as I know, `log4rs` does not support `tracing`. You should to implement a middleware(like [Tracing](https://github.com/poem-web/poem/blob/master/poem/src/middleware/tracing_mw.rs)) yourself and call the `log4rs`'s API for recording.
